### PR TITLE
Switch base image to UBI for AMD rocm install

### PR DESCRIPTION
**Pull request by harshad16**
## Description

Switch base image to UBI for AMD rocm install
Related-to: https://issues.redhat.com/browse/RHOAIENG-7501

## How Has This Been Tested?

```
1. build the base `podman build -t amd-base .`
2. check the necessary bits are available in amd-base , with `rpm -qa`
```

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by jstourac** [comment]
Looks like we need to update the prow configuration then too:
* https://github.com/opendatahub-io/notebooks/actions/runs/9943091202/job/27466099229?pr=620
* https://github.com/opendatahub-io/notebooks/actions/runs/9943091202/job/27466099489?pr=620

**Review by jstourac — APPROVED** [review]

File: amd/ubi9-python-3.9/Dockerfile
Comment (caponetto) [inline amd/ubi9-python-3.9/Dockerfile:23]: Is there any particular reason for not combining sequential `RUN` commands into a single one? I understand each `RUN` creates one extra layer, which extends build time and increases the container image size.

Comment (jiridanek) [reply amd/ubi9-python-3.9/Dockerfile:23]: If your `yum install` installs new files into `/etc/yum.repos.d/`, then there need to be multiple yum invocations, one to install the new .repo files, and the other one to use the repos. Still, it could still be done in a single Dockerfile `RUN`.

If the only reason for multiple RUNs is readability, would it make sense to put all these RUNs into a single bash script and run that from the Dockerfile? Another reason may be faster and simpler local development that makes use of layer caching.

This image is IMO ok, when there are multiple `yum installs` in some other images where each downloads the yum cache, deletes it and the end, and the yum running right after downloads cache again, that's something to fix!

**Comment by jiridanek** [comment]
Also Makefile should be updated (in this PR), to reference ubi9- and not c9s- dir.

File: amd/ubi9-python-3.9/Dockerfile
Comment (atheo89) [inline amd/ubi9-python-3.9/Dockerfile:31]: Just a suggestion if this makes sense to you:

Could we place the ROCm and AMDGPU RPMs in external files and then use the COPY  to include them in the Dockerfile?

File: amd/ubi9-python-3.9/Dockerfile
Comment (atheo89) [inline amd/ubi9-python-3.9/Dockerfile:1]: As mentioned we need also this on the Makefile:

```
.PHONY: amd-ubi9-python-3.9
amd-ubi9-python-3.9: base-ubi9-python-3.9
	$(call image,$@,amd/ubi9-python-3.9,$<)
```

**Review by atheo89 — COMMENTED** [review]
Great transition to UBI :slightly_smiling_face: 
I added some minor suggestions, overall looks good to me.

I was able to built this image you may find it here:
`quay.io/rh_ee_atheodor/workbench-images:amd-ubi9-python-3.9-2024a_20240717`

File: amd/ubi9-python-3.9/Dockerfile
Comment (harshad16) [reply amd/ubi9-python-3.9/Dockerfile]: >Is there any particular reason for not combining sequential RUN commands into a single one? I understand each RUN creates one extra layer, which extends build time and increases the container image size.

Rightly pointed out, corrected it to the single RUN command

**Comment by jiridanek** [comment]
We're still running out of disk space on one of the images, as @caponetto observed yesterday.

**Review by jiridanek — APPROVED** [review]

**Comment by jiridanek** [comment]
>  ci/prow/notebook-amd-c9s-python-3-9-pr-image-mirror — Job failed.                     BaseSHA:fecd10c13ce13d66b57498abc48976f74121dd63

so openshift-ci needs to be updated

**Comment by jiridanek** [comment]
@jstourac so, approving in GitHub UI gives BOTH LGTM and approved, now that we are approvers. Gotta be careful.

File: Makefile
Comment (atheo89) [inline Makefile:276]: Maybe we need to remove this comment now, to avoid confusion.

**Comment by caponetto** [comment]
> We're still running out of disk space on one of the images, as @caponetto observed yesterday.

At first, I thought that was happening only during Trivy scan because it has to copy stuff around. But then I saw that CI is running out of disk space even during the build step for `amd-jupyter-pytorch-ubi9-python-3.9` (examples are this PR and some builds triggered after a merge). Considering that `amd-jupyter-pytorch-ubi9-python-3.9` is ~60 GB uncompressed, the CI is operating on its limit. If we ever need to add new things to this image, we'll probably face these storage issues more often.

**Comment by atheo89** [comment]
To fix the ci please check this PR:  https://github.com/opendatahub-io/notebooks/pull/627 explain the instructions on how you can update the notebook matrix

File: ci/cached-builds/gen_gha_matrix_jobs.py
Comment (jiridanek) [reply ci/cached-builds/gen_gha_matrix_jobs.py:202]: @atheo89 I suggest we merge it without making the change, and I can do this later, what do you think?

This test does not run in the CI, so we won't spoil anything. And it would be good to start merging the rocm work soon, because the time is running short.

Comment (jiridanek) [reply ci/cached-builds/gen_gha_matrix_jobs.py:202]: When this and the other groundwork gets merged, we can seriously try to decrease image size by removing the unneeded parts of the sdk that are not required to run pytorch/tensorflow. It would be best not to try to release with full sdk, these huge images are too hard to work with.

**Comment by atheo89** [comment]
/lgtm 
/approve

/override ci/prow/images 
/override /ci/prow/notebook-rocm-jupyter-pyt-ubi9-python-3-9-pr-image-mirror 
/override ci/prow/notebook-rocm-jupyter-tf-ubi9-python-3-9-pr-image-mirror
/override ci/prow/rocm-notebooks-e2e-tests

**Comment by jiridanek** [comment]
let me try

/override "build (rocm-jupyter-pytorch-ubi9-python-3.9) / build"
/override ci/prow/notebook-rocm-jupyter-tf-ubi9-python-3-9-pr-image-mirror
/override ci/prow/rocm-notebooks-e2e-tests
/override ci/prow/notebook-rocm-jupyter-pyt-ubi9-python-3-9-pr-image-mirror
/override ci/prow/images

**Comment by jiridanek** [comment]
/override "build (rocm-jupyter-pytorch-ubi9-python-3.9) / build"

**Pull request merged**
